diff --git a/contextual_help.module b/contextual_help.module
index b4343e6..c1cb42b 100755
--- a/contextual_help.module
+++ b/contextual_help.module
@@ -304,6 +304,10 @@ function contextual_help_views_pre_render(&$views) {
     // Include the CTools tools that we need.
     ctools_include('ajax');
     ctools_include('modal');
+	
+	 // Add CTools' javascript to the page.
+	ctools_modal_add_js();
+	
     // Create our own javascript that will be used to theme our modals.
     $contextual_help_style = array(
       'contextual-help-modal-style' => array(
@@ -321,7 +325,7 @@ function contextual_help_views_pre_render(&$views) {
           'background-color' => '#FFF',
         ),
         'animationSpeed' => 'fast',
-        'modalTheme' => 'CToolsModalDialog',
+        'modalTheme' => 'ContextualHelpModal',
         'throbberTheme' => 'CToolsModalThrobber',
       ),
     );
diff --git a/js/contextual_help.js b/js/contextual_help.js
index 7482a2b..9ec997b 100755
--- a/js/contextual_help.js
+++ b/js/contextual_help.js
@@ -3,7 +3,7 @@
 * Provide the HTML to create the modal dialog.
 */
 (function ($) {
-Drupal.theme.prototype.contextual_help_modal = function () {
+Drupal.theme.prototype.ContextualHelpModal = function () {
   var html = '';
   html += '<div id="ctools-modal" class="popups-box">';
   html += '  <div class="ctools-modal-content ctools-modal-contextual-help-modal-content">';
